source/server/config/network/http_connection_manager.h

```diff
@@ -48,10 +49,19 @@ class HttpFilterConfigFactory {
 public:
   virtual ~HttpFilterConfigFactory() {}
 
-  virtual HttpFilterFactoryCb tryCreateFilterFactory(HttpFilterType type, const std::string& name,
-                                                     const Json::Object& config,
-                                                     const std::string& stat_prefix,
-                                                     Server::Instance& server) PURE;
+  /**
+  * Create a particular http filter factory implementation.  If the creation fails, an error will be
+  * thrown.  The returned callback should always be valid.
+  */
+  virtual HttpFilterFactoryCb createFilterFactory(HttpFilterType type, const Json::Object& config,
+                                                  const std::string& stat_prefix,
+                                                  Server::Instance& server) PURE;
+
+  /**
+  * Provides the identifying name for a particular implementation of an http filter produced by the
+  * factory.
+  */
+  virtual std::string name() PURE;
```

Comment:
I was curious about this.  We would need to define a string as a member of the class or make a static string, right?  If I'm thinking about it correctly, both of those options would ultimately be more copying than the current version (because we should only call the method once during registration and the string produced will be moved into the unordered_map as opposed to the copy required for a const std::string &).  However, if this is just a readability/API preference for envoy, that totally makes sense.